app/src/main/java/com/antosik/benchproject/app/popular/movies/viewModel/PopularMoviesViewModel.kt

```diff
@@ -17,45 +17,33 @@ class PopularMoviesViewModel @Inject constructor(
     private val getPopularMoviesUseCase: GetPopularMoviesUseCase,
     private val popularMoviesFragmentNavigator: PopularMoviesFragmentNavigator,
 ) : ViewModel() {
+    val popularMoviesViewState: LiveData<UIState>
+        get() = _popularMoviesViewState
 
-    val popularMovies: LiveData<List<Movie>>
-        get() = _popularMovies
-    val isLoaderVisible: LiveData<Boolean>
-        get() = _isLoaderVisible
-    val isPlaceholderVisible: LiveData<Boolean>
-        get() = _isPlaceholderVisible
-    val isRefreshing: LiveData<Boolean>
-        get() = _isRefreshing
-
-    private val _popularMovies = MutableLiveData<List<Movie>>(emptyList())
-    private val _isLoaderVisible = MutableLiveData<Boolean>()
-    private val _isPlaceholderVisible = MutableLiveData<Boolean>()
-    private val _isRefreshing = MutableLiveData<Boolean>()
+    private val _popularMoviesViewState = MutableLiveData<UIState>()
 
     init {
-        loadPopularMovie()
+        loadPopularMovies()
     }
 
     fun navigateToMovieDetailsFragment(movieId: Int) {
         popularMoviesFragmentNavigator.navigateToMovieDetailsFragment(movieId)
     }
 
     fun onRefreshPopularMovies() {
-        loadPopularMovie()
```

Comment:
here is also some logic that should be tested :) for now, i would go just for one additional test, if the use case was triggered when onRefreshPopularMovies is called, because this can be maybe improved with more dynamic approach, but i don't have a proper idea now :D

Reply:
Done 😄